### PR TITLE
Doc of auto schema change for mysql cdc

### LINKED ISSUE #2489: Document: feat(cdc): auto schema change for mysql cdc
This issue tracks the documentation update needed for the merged PR #17876.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/17876
Source PR Merged At: 2024-08-20T09:21:49Z

If it is a major improvement that deserves a new page or a new section in the documentation, please check if we should label it as an experiment feature.

### PR DESCRIPTION
**Pull request by WanYixian**
## Description

Doc of auto schema change for mysql cdc, @StrikeW pls help check the accuracy of words and code, and let me know if any comments, thanks!

## Related code PR

https://github.com/risingwavelabs/risingwave/pull/17876
## Related doc issue

Resolve https://github.com/risingwavelabs/risingwave-docs/issues/2489

## Checklist

- [ ] I have checked the doc site preview, and the updated parts look good.
- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: src/pages/product-lifecycle.md [hunk 1]
```diff
@@ -23,18 +23,18 @@ As introduced above, when you see a "Public preview" note in the documentation,
 
 The following is a list of all features in the public preview phase:
 
-| Feature name                            | Start date | Start version |
-|-----------------------------------------|------------|---------------|
-| [Subscription](/docs/next/subscription) | 2024.5     | 1.9           |
-| [Time travel queries](/docs/next/time-travel-queries/) | 2024.7     | 2.0           |
-| [Manage secrets](/docs/next/manage-secrets/) | 2024.7     | 2.0           |
-| [Decouple sinks](/docs/next/data-delivery/#sink-decoupling) | 2023.10    | 1.3           |
+| Feature Name            | Start Date | Start Version |
```
Comment (ShanlanLi) [inline src/pages/product-lifecycle.md]: Why use the title case here? We use sentence case.

Comment (WanYixian) [reply src/pages/product-lifecycle.md]: fixed

File: docs/guides/ingest-from-mysql-cdc.md [hunk 2]
```diff
@@ -437,6 +437,58 @@ And this it the output of `DESCRIBE supplier;`
...
+This feature is only available in the premium edition of RisingWave. The premium edition offers additional advanced features and capabilities beyond the free and community editions. If you have any questions about upgrading to the premium edition, please contact our sales team at [sales@risingwave-labs.com](mailto:sales@risingwave-labs.com).
+:::
+
+:::info Public Preview
+This feature is in the public preview stage, meaning it's nearing the final product but is not yet fully stable. If you encounter any issues or have feedback, please contact us through our [Slack channel](https://www.risingwave.com/slack). Your input is valuable in helping us improve the feature. For more information, see our [Public preview feature list](/product-lifecycle/#features-in-the-public-preview-stage).
+:::
+
+RisingWave supports auto schema changes in MySQL CDC. It ensures that your RisingWave pipeline stays synchronized with any schema changes in the source database, reducing the need for manual updates and preventing inconsistencies.
+
+Currently, RisingWave supports the `ALTER TABLE` command to add or drop columns. We plan to add support for other DDL operations in future releases.
```
Comment (StrikeW) [inline docs/guides/ingest-from-mysql-cdc.md]: I think we should make the capability of this feature more **clear** in the doc. e.g. list the supported operations 
- `ADD COLUMN [DEFAULT expr]`, only constant value expression is supported for the default value.
- `DROP COLUMN`

File: docs/guides/ingest-from-mysql-cdc.md [hunk 3]
```diff
@@ -437,6 +437,62 @@ And this it the output of `DESCRIBE supplier;`
...
+Add columns to the MySQL table and observe the changes in RisingWave:
+
+```sql
+-- In MySQL:
+USE mytest;
+ALTER TABLE customers ADD COLUMN v1 VARCHAR(255);
+ALTER TABLE customers ADD COLUMN v2 DOUBLE(5,2);
+
+-- In RisingWave (after a brief pause):
+DESCRIBE rw_customers;
```
Comment (StrikeW) [inline docs/guides/ingest-from-mysql-cdc.md:493]: could be better if we can show the output of `describe` as well.

**Review by StrikeW — APPROVED** [review]
Rest LGTM!

File: docs/guides/ingest-from-mysql-cdc.md [hunk 4]
```diff
@@ -437,6 +437,62 @@ And this it the output of `DESCRIBE supplier;`
...
+
+:::info Public Preview
+This feature is in the public preview stage, meaning it's nearing the final product but is not yet fully stable. If you encounter any issues or have feedback, please contact us through our [Slack channel](https://www.risingwave.com/slack). Your input is valuable in helping us improve the feature. For more information, see our [Public preview feature list](/product-lifecycle/#features-in-the-public-preview-stage).
+:::
+
+RisingWave supports auto schema changes in MySQL CDC. It ensures that your RisingWave pipeline stays synchronized with any schema changes in the source database, reducing the need for manual updates and preventing inconsistencies.
+
+Currently, RisingWave supports the `ALTER TABLE` command with the following operations, and we plan to add support for additional DDL operations in future releases.
+
+- `ADD COLUMN [DEFAULT expr]`: Allows you to add a new column to an existing table, only constant value expressions is supported for the default value.
```
Comment (ShanlanLi) [inline docs/guides/ingest-from-mysql-cdc.md]:
```suggestion
- `ADD COLUMN [DEFAULT expr]`: Allows you to add a new column to an existing table. Only constant value expressions are supported for the default value.
```

**Review by ShanlanLi — APPROVED** [review]
LGTM

**Pull request merged**